test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ScalaJSDefinedTest.scala

```diff
@@ -565,6 +568,32 @@ class ScalaJSDefinedTest {
     assertEquals(100, dyn.foo())
   }
 
+  @Test def readonly_properties(): Unit = {
+    assumeFalse(
+        "Assuming strict mode semantics, which are not honored by Rhino",
+        Platform.executingInRhino)
+
+    // Named classes
+    @ScalaJSDefined
+    class Foo extends js.Object {
+      def bar: Int = 1
```

Comment:
Just to check my understanding: You are not testing for vals, because we cannot make them immutable.